cmd/nvidia-dra-controller/imex.go

```diff
@@ -249,10 +246,13 @@ func (m *ImexManager) streamImexDomains(ctx context.Context) (<-chan string, <-c
 }
 
 // generateImexChannelPool generates the contents of a ResourceSlice pool for a given IMEX domain.
-func generateImexChannelPool(imexDomain string, numChannels int) resourceslice.Pool {
-	// Generate dchannels from 0 to numChannels
+func generateImexChannelPool(imexDomain string, imexChannelOffsets map[string]map[string]int) resourceslice.Pool {
```

Comment:
As I mentioned in my comments yesterday, I still think this should function should take 3 values -- an imex domain string, a start channel, and a count of how many channels to include. Any math needed to figure that out and pass those values in should be handled external to this function.